### PR TITLE
Updates react-mosaic and makes WindowTopBar draggable

### LINKED ISSUE #1856: Mosaic mode --> draggable windows
We are currently able to resize windows, but unable to arrange where they appear in the workspace. `react-mosaic` allows us to drag windows around and dynamically split other windows.

mosaic-window-split

A first stab at this work would involve analyzing the [react-mosaic docs](https://github.com/palantir/react-mosaic) and reporting back to the team re: feasibility. 

### Out of Scope
- Other `react-mosaic` functionality: (e.g. 
  - `replace`, `split`, `auto-arrange` etc. (will loop back w/ design team)
  - `expand` (Possibly related to #1703)

### Resources
- [Interactive designs](https://github.com/ProjectMirador/mirador-design/wiki/Mirador-3-Interactive-Designs)
- Mosaic mode design #1788

### PR DESCRIPTION
**Pull request by mejackreed**
Fixes #1856 

draggabletopbar

Spin off work (out of scope for this PR):
 - #1981 custom preview work

### CODE REVIEW

File: src/components/Window.js [hunk 1]
```diff
@@ -10,19 +10,36 @@ import ThumbnailNavigation from '../containers/ThumbnailNavigation';
  * @param {object} window
  */
 export class Window extends Component {
+  /**
+   * wrappedTopBar - will conditionally wrap a WindowTopBar for needed
+   * additional functionality based on workspace type
+   */
+  wrappedTopBar() {
```
Comment (mejackreed) [inline src/components/Window.js]: Open to other suggestions on the best way to do this. We essentially need to conditionally wrap this component.

File: src/components/WorkspaceMosaic.js [hunk 2]
```diff
@@ -69,10 +69,24 @@ export class WorkspaceMosaic extends React.Component {
...
-      <Window
-        key={window.id}
-        window={window}
-      />
+      <MosaicWindow
+        toolbarControls={[]}
+        additionalControls={[]}
+        path={path}
+        windowId={window.id}
+        renderPreview={() => (
```
Comment (mejackreed) [inline src/components/WorkspaceMosaic.js]: A better styled preview spun off into #1981

File: src/components/WorkspaceMosaic.js [hunk 3]
```diff
@@ -69,10 +69,24 @@ export class WorkspaceMosaic extends React.Component {
...
+          <div className="mosaic-preview">
+            <div className="mosaic-window-body">
+              <h4>Mirador</h4>
```
Comment (jkeck) [inline src/components/WorkspaceMosaic.js]: Can we i18n this?

Comment (mejackreed) [reply src/components/WorkspaceMosaic.js]: Good call. Just pushed a commit to do this.

**Pull request merged**